src/pages/modifyManage/ModifyManage.tsx

```
@@ -35,138 +39,203 @@ import * as S from "./ModifyManage.styled";
 import { isAllFieldsFilled } from "./utils/handleEvent";
 
-const ModifyManage = () => {
-  const [modifyManageStep, setModifyManageStep] = useState(1); // 등록 step 나누기
-  const { openConfirm, closeConfirm, openAlert, closeAlert } = useModal();
+// Reducer로 상태 관리 통합
+type State = {
+  performanceTitle: string;
+  genre: SHOW_TYPE_KEY;
+  runningTime: number | null;
+  performanceDescription: string;
+  performanceAttentionNote: string;
+  accountNumber: string;
+  posterImage: string;
+  performanceTeamName: string;
+  performanceVenue: string;
+  performancePeriod: string;
+  performanceContact: string;
+  ticketPrice: number | null;
+  totalScheduleCount: number;
+  scheduleList: Schedule[];
+  castList: Cast[];
+  staffList: Staff[];
+  bankName: BANK_TYPE;
+  accountHolder: string;
+};
 
+type ModifyState = {
+  ModifyManageStep: number;
+  isBookerExist: boolean | undefined;
+  isFree: boolean;
+  isChecked: boolean;
+  bankOpen: boolean;
+};
+
+type Action =
+  | { type: "SET_DATA"; payload: Partial<State> }
+  | { type: "SET_FIELD"; field: keyof State; value: State[keyof State] }
+  | { type: "SET_SCHEDULE_COUNT"; payload: number };
+
+const initialState: State = {
+  performanceTitle: "",
+  genre: "ETC",
+  runningTime: null,
+  performanceDescription: "",
+  performanceAttentionNote: "",
+  accountNumber: "",
+  posterImage: "",
+  performanceTeamName: "",
+  performanceVenue: "",
+  performancePeriod: "",
+  performanceContact: "",
+  ticketPrice: null,
+  totalScheduleCount: 1,
+  scheduleList: [],
+  castList: [],
+  staffList: [],
+  bankName: "NONE",
+  accountHolder: "",
+};
+
+const reducer = (state: State, action: Action): State => {
+  switch (action.type) {
+    case "SET_DATA":
+      return { ...state, ...action.payload };
+    case "SET_FIELD":
+      return { ...state, [action.field]: action.value };
+    case "SET_SCHEDULE_COUNT":
+      return { ...state, totalScheduleCount: action.payload };
+    default:
+      return state;
+  }
+};
```

p3) 오오👍
reducer에서 default로는 `throw new Error` 와 같이 에러를 많이 던진다고 해요! 
왜냐하면, type에 맞지 않는 값이 dispatch에 들어가면 오류니까요! 

지금도 너무 잘 작성해주셨는데, useReducer를 더 잘 쓰는 방법과 관련해서 아티클 하나 남길게요!
https://devtrium.com/posts/how-to-use-react-usereducer-hook#multiple-pieces-of-state-that-rely-on-each-other

---

아티클 살펴봤는데 좋은 내용이 많네요!

**렌더링 사이에 set 함수(dispatch)의 참조가 바뀌지 않는 것을 보장한다.**
(React guarantees that its identity won't change between renders.)

**부정확한 액션 타입이 전달된 경우, 원하지 않은 동작을 하는 것이니 에러 처리를 하는 것이 더 좋다.**
아니라면, 눈에 띄지 않는 버그가 되니 반드시 디버깅의 용이성을 위해 throw 하자
(This is basically creating a [silent error](https://en.wikipedia.org/wiki/Error_hiding) when an incorrect action type is provided, and silent errors can be very hard to debug.)

게으른 초기화를 위해서 state 함수에 콜백함수를 넣어주듯, **dispatch의 3번째 인자를 활용해서 게으른 초기화**를 통해 복잡한 계산 시 성능 향상을 도모할 수 있다.
(Lazy initialization)

이 정도가 핵심 내용이었던 것 같은데, 본 김에 throw로 수정하도록 하겠습니다! 
감사합니다 😊